models/DataObject/ClassDefinition.php

```diff
@@ -208,11 +209,25 @@ final class ClassDefinition extends Model\AbstractModel implements ClassDefiniti
      */
     public static function getById(string $id, bool $force = false): ?ClassDefinition
     {
+        $cacheKey = 'class_' . $id;
+        $class = null;
+
+        if(!$force &&
+            RuntimeCache::isRegistered($cacheKey)
+        ) {
+            $class = RuntimeCache::get($cacheKey);
+        }
+
+        if($class !== null) {
+            return $class;
+        }
+
         $class = (new ClassDefinition\Listing())
+            ->setForce($force)
```

Comment:
i believe we have to set
```suggestion
            ->setForce(true)
```
otherwise, in the case where `$force` is `false` and the RuntimeCache is returning `null`, it would end up in the Listing and check the RuntimeCache again (which we already know to be nullish)

The rest LGTM

Reply:
the slight difference is that it  would be throwing https://github.com/pimcore/pimcore/blob/8b9ba3f58f0bfc331ac6cfd2f7ed89f5c4af691d/models/DataObject/ClassDefinition/Listing/Dao.php#L72
instead of checking again the runtime cache and throw https://github.com/pimcore/pimcore/blob/8b9ba3f58f0bfc331ac6cfd2f7ed89f5c4af691d/models/DataObject/ClassDefinition/Listing/Dao.php#L76
but not a big deal